openslides/core/static/js/core/start.js

```diff
@@ -46,6 +47,27 @@ angular.module('OpenSlidesApp.core.start', [])
                 this.bootup();
             },
         };
+
+        // We need to 'ping' the server with a get request to whoami, because then we can decide,
+        // if the server is down or respond with a 403 (this cannot be differentiated with websockets)
+        autoupdate.registerRetryConnectCallback(function () {
+            return $q(function (resolve) {
```

Comment:
If you return any value inside the successCallback or errorCallback of `then()`, the new promise will be resolved. If you throw an error, the new promise will be rejected.

See https://docs.angularjs.org/api/ng/service/$q#the-promise-api and https://developers.google.com/web/fundamentals/getting-started/primers/promises#chaining